src/Core/Workspace/Project.cs

```diff
@@ -37,6 +37,46 @@ internal class ProjectFileComparer : EqualityComparer<Project>
             project.ProjectFile.GetHashCode(StringComparison);
     }
 
+    /// <summary>
+    /// List of arguments that are part of the project loaded event.
+    /// </summary>
+    public class ProjectLoadedEventArgs : EventArgs
+    {
+        public ProjectLoadedEventArgs(Uri? projectUri, int sourceFileCount, int projectReferenceCount, int packageReferenceCount, TimeSpan duration)
```

Comment:
Is there a way to track if a project load happened explicitly, or as the result of a transitive dependency from another load event?

Reply:
Yes, that's a good idea, I'll add a `UserAdded` field here, which will track whether the project was added directly via `%project` or `qsharp.projects.add()`.